workspaces/bookmarks/plugins/bookmarks/src/plugin.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createPlugin,
+  createRoutableExtension,
+} from '@backstage/core-plugin-api';
+
+import { entityRootRouteRef } from './routes';
+
+/**
+ * Bookmarks plugin
+ *
+ * @public
+ */
+export const bookmarksPlugin = createPlugin({
+  id: 'bookmarks',
+  routes: {
+    entityRoot: entityRootRouteRef,
+  },
+});
+
+/**
+ * BookmarksTab extension for the contents of the Bookmarks tab
+ *
+ * @public
+ */
+export const BookmarksTab = bookmarksPlugin.provide(
+  createRoutableExtension({
+    name: 'BookmarksTab',
+    component: () =>
+      import('./components/BookmarksTab/BookmarksTab').then(
+        m => m.BookmarksTab,
+      ),
+    mountPoint: entityRootRouteRef,
+  }),
+);
```

Comment:
The alternative to `createRoutableExtension` is maybe `createComponentExtension`. As said. Not sure if you like to have sub-routes for the selected bookmark in the future?!

```tsx
export const EntityBookmarksContent = bookmarksPlugin.provide(
  createComponentExtension({
    name: 'EntityBookmarksContent',
    component: {
      lazy: () =>
        import('./components/EntityBookmarksContent/EntityBookmarksContent').then(
          m => m.EntityBookmarksContent,
        ),
    },
  }),
);
```

Reply:
I also saw this done by the `todo` plugin and followed them:

https://github.com/backstage/community-plugins/blob/6464668e46fa93a3e99dbd1eaa186b2a637fe15f/workspaces/todo/plugins/todo/src/plugin.ts#L50-L61

I am not sure if this is needed for sub-routes either, but as said, I think that's a great idea, so I like having the idea of having "room to grow" per se :)